packages/react-native/src/primitives/FederatedProviderButton/FederatedProviderButton.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { useCallback } from 'react';
+import {
+  PressableStateCallbackType,
+  StyleProp,
+  View,
+  ViewStyle,
+} from 'react-native';
+
+import { Button, Label } from '..';
+
+import { styles } from './styles';
+import { FederatedProviderButtonProps } from './types';
+
+export default function FederatedProviderButton({
```

Comment:
Wondering if this is a primitive or if it is Authenticator specific.  Thinking this should be an authenticator component.

Reply:
This was part of the ask since we don't have a directory for common components yet. Now thinking we can just put it in _src/Authenticator/Defaults/common/_ since I've had more time to think through it

Reply:
Yes, this will be moved into the `./Authenticator` folder once the [template PR](https://github.com/aws-amplify/amplify-ui/pull/2769) gets merged